### PR TITLE
Issue 67 inspirations google

### PR DESCRIPTION
**Pull request by jvaughn575**
This is an incredible feature, thanks to the power of Google maps. So you can search for anyplace in the world with the autocomplete search bar. Then you can drag the yellow pinman on the bottom right of the map on to the map. Dots represent user content images and pano, and highlighted roads are google streetview areas. So after you drop pinman on to the map at one of the highlighted map areas, you will be transported there via streetview, then you can look around, and when you hit the camera but a modal will pop up of the inspiration you started. You can change the default description if you don't like it and hit add. It will save it as an inspiration and display it on the inspiration page. 

I know this is alot in one pr, but just creating inspirations from google streetview isn't that could if you can't see them. So I had to fix the inspiration page as well.

### CODE REVIEW

File: src/components/Pages/Explore/BookmarkMapInspiration/BookmarkMapInspiration.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            message.error("Inspiration was not saved!");
+          }
+        });
+      this.setState({
+        loading: true,
+        imageChosen: false,
+        selectedImageAttrs: null,
+        title: ""
+      });
+      setTimeout(() => {
```
Comment (reggieroby) [inline src/components/Pages/Explore/BookmarkMapInspiration/BookmarkMapInspiration.js]: probably should try not to use setTimeout's like this.

Comment (reggieroby) [reply src/components/Pages/Explore/BookmarkMapInspiration/BookmarkMapInspiration.js]: im not sure of how id change it though :) .

File: src/components/Pages/Explore/GoogleMap.js [hunk 2]
```diff
@@ -71,47 +87,88 @@ export const CustomMap = compose(
...
+      </SearchBox>        
+      {props.markers.map((marker, index) =>
+        <Marker key={index} position={marker.position} />
+        
+      )}
+    </GoogleMap>
+
+    <Button       
+      style = {{minHeight: 60}}
+      onClick = {() => {
```
Comment (reggieroby) [inline src/components/Pages/Explore/GoogleMap.js]: you should take this out and name it.

File: src/components/Pages/Explore/GoogleMap.js [hunk 3]
```diff
@@ -71,47 +87,88 @@ export const CustomMap = compose(
...
-      />
-    </SearchBox>
-    {props.markers.map((marker, index) =>
-      <Marker key={index} position={marker.position} />
-    )}
-  </GoogleMap>
+      <Icon style={{fontSize: 40 }} type="camera" /> 
+    </Button>
+
+    {props.isModalOpen && <BookmarkMapInspiration
```
Comment (reggieroby) [inline src/components/Pages/Explore/GoogleMap.js]: i like angular ngif better :/

File: src/components/Pages/Inspiration.js [hunk 4]
```diff
@@ -39,25 +42,78 @@ class PinInput extends Component {
...
       style={{ width: 240 }}
-      cover={<img src="https://assets.atlasobscura.com/media/W1siZiIsInVwbG9hZHMvcGxhY2VfaW1hZ2VzL2RlODZjMTUyZWY2YWRlZmYxNDljNWIxNzU2NjNmYThhNzI4NTVhNzMuanBnIl0sWyJwIiwidGh1bWIiLCI5ODB4PiJdLFsicCIsImNvbnZlcnQiLCItcXVhbGl0eSA4MSAtYXV0by1vcmllbnQiXV0/de86c152ef6adeff149c5b175663fa8a72855a73.jpg" alt="" />}
+      cover={<img src={image || "https://assets.atlasobscura.com/media/W1siZiIsInVwbG9hZHMvcGxhY2VfaW1hZ2VzL2RlODZjMTUyZWY2YWRlZmYxNDljNWIxNzU2NjNmYThhNzI4NTVhNzMuanBnIl0sWyJwIiwidGh1bWIiLCI5ODB4PiJdLFsicCIsImNvbnZlcnQiLCItcXVhbGl0eSA4MSAtYXV0by1vcmllbnQiXV0/de86c152ef6adeff149c5b175663fa8a72855a73.jpg"} alt="" />}
     >
       <Meta
-        title="Kasbah du Toubkal"
-        description={<a href="https://www.atlasobscura.com/places/kasbah-du-toubkal" rel="noopener noreferrer" target="_blank">Source</a>}
+        title={description || "Kasbah du Toubkal"} 
+
+        /* IMPORTANT! need to add href attribution of images before site is live to public */
```
Comment (reggieroby) [inline src/components/Pages/Inspiration.js]: this would be a spot to add `TODO: ....`. use them sparingly.

Comment (reggieroby) [reply src/components/Pages/Inspiration.js]: then you can search for TODO and IDE's highlight it.

File: src/userApi.js [hunk 5]
```diff
@@ -1,4 +1,6 @@
 import React, { Component } from 'react';
+import { app } from './components/app';
+import { saveState } from './models/localStorage';
 
 const api = process.env.REACT_APP_SERVER_URL || 'http://localhost:3001/api';
```
Comment (reggieroby) [inline src/userApi.js]: this should be elevated to the config file.

File: src/components/Pages/Inspiration.js (hunk 4 above)
Comment (jvaughn575) [reply src/components/Pages/Inspiration.js]: Ok, yeah that would be easier to search.

File: src/userApi.js (hunk 5 above)
Comment (jvaughn575) [reply src/userApi.js]: How do I do that, I am not quite sure what I should do?

**Pull request merged**